jme3-core/src/plugins/java/com/jme3/export/binary/ByteUtils.java

```diff
@@ -479,9 +479,8 @@ public static byte[] readData(byte[] store, int bytes, InputStream is) throws IO
     public static byte[] rightAlignBytes(byte[] bytes, int width) {
         if (bytes.length != width) {
             byte[] rVal = new byte[width];
-            for (int x = width - bytes.length; x < width; x++) {
-                rVal[x] = bytes[x - (width - bytes.length)];
-            }
+            if (width - (width - bytes.length) >= 0)
+                System.arraycopy(bytes, width - bytes.length - (width - bytes.length), rVal, width - bytes.length, width - (width - bytes.length));
```

Comment:
Please simplify `width - bytes.length - (width - bytes.length)` to `0` and `width - (width - bytes.length)` to `bytes.length`.